providers/implementations/encode_decode/encode_key2any.c

```diff
@@ -817,6 +818,58 @@ static int ecx_pki_priv_to_der(const void *vecxkey, unsigned char **pder)
 
 /* ---------------------------------------------------------------------- */
 
+#ifndef OPENSSL_NO_ML_DSA
+static int ml_dsa_spki_pub_to_der(const void *vkey, unsigned char **pder)
+{
+    const ML_DSA_KEY *key = vkey;
+    size_t publen;
+
+    if (ossl_ml_dsa_key_get_pub(key) == NULL) {
+        ERR_raise(ERR_LIB_PROV, ERR_R_PASSED_NULL_PARAMETER);
+        return 0;
+    }
+    publen = ossl_ml_dsa_key_get_pub_len(key);
+
+    if (pder != NULL
+            && ((*pder = OPENSSL_memdup(ossl_ml_dsa_key_get_pub(key),
+                                        publen)) == NULL))
+        return 0;
+
+    return publen;
+}
+
+static int ml_dsa_pki_priv_to_der(const void *vkey, unsigned char **pder)
+{
+    const ML_DSA_KEY *key = vkey;
+    size_t len;
+
+    if (ossl_ml_dsa_key_get_priv(key) == NULL) {
+        ERR_raise(ERR_LIB_PROV, ERR_R_PASSED_NULL_PARAMETER);
+        return 0;
+    }
+    len = ossl_ml_dsa_key_get_priv_len(key);
+
+    if (pder != NULL
+            && ((*pder = OPENSSL_memdup(ossl_ml_dsa_key_get_priv(key), len)) == NULL))
+        return 0;
```

Comment:
This is currently not octet-string wrapped, ... just raw bytes. We'll review in the next PR.